Parser/ExprGrapher.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace PSI;
+class ExprGrapher : Visitor<int> {
+   StringBuilder mSB = new ();
+   int mID = 1;
+
+   public override int Visit (NLiteral literal) {
+      mSB.AppendLine ($"Id{mID}[{literal.Value.Text}\\n{literal.Type}]");
+      return mID++;
+   }
+
+   public override int Visit (NIdentifier identifier) {
+      mSB.AppendLine ($"Id{mID}[{identifier.Name.Text}\\n{identifier.Type}]");
+      return mID++;
+   }
+
+   public override int Visit (NUnary unary) {
+      int down = unary.Expr.Accept (this);
+      int opID = mID;
+      mSB.AppendLine ($"Id{mID++}([{unary.Op.Text}\\n" +

```

Comment:
I think implementing ExprTyper and then adding the type to the grapher output will be better. In that way we can call unary.Type here. Sorry for confusing. If you can, please roll back the type related changes.

Reply:
I implemented all of it in a single commit, I'll implement Typer and re-assign